### PR TITLE
Implements filtering the `sorting-tree.component`

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**
- Added filtering by visibility, like internal, public or hidden items
- Added option to collapse or expand all nodes
- Added also style for the dark theme

### CODE REVIEW

File: client/src/app/core/ui-services/tree.service.ts [hunk 1]
```diff
@@ -46,6 +46,8 @@ export interface FlatNode {
     isSeen: boolean;
     expandable: boolean;
     id: number;
+    tag?: any;
```
Comment (FinnStutzenstein) [inline client/src/app/core/ui-services/tree.service.ts]: What is tag used for?

Comment (GabrielInTheWorld) [reply client/src/app/core/ui-services/tree.service.ts]: `tag` is used for something a component can search for or to filter the node(s).

File: client/src/app/shared/components/sorting-tree/sorting-tree.component.html [hunk 2]
```diff
@@ -36,10 +36,12 @@
                         chevron_right
                     </mat-icon>
                 </button>
-                {{ node.name }}
+                <ng-container
+                    [ngTemplateOutlet]="innerNode"
+                    [ngTemplateOutletContext]="{node: node}"></ng-container>
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/sorting-tree/sorting-tree.component.html]: `item: node.item`. Do not make internals public.

File: client/src/app/site/agenda/components/agenda-sort/agenda-sort.component.html [hunk 3]
```diff
@@ -7,13 +7,59 @@
...
+        (visibleNodes)="onChangeAmountOfItems($event)"
         [model]="itemsObservable"
+        [stateChange]="changeState"
+        [filterChange]="changeFilter"
         parentKey="parent_id"
         weightKey="weight"
-    ></os-sorting-tree>
+    >
+        <ng-template #innerNode class="sorting-tree-node" let-node="node">
+            <span>{{ node.item.getTitle() }}</span>
```
Comment (FinnStutzenstein) [inline client/src/app/site/agenda/components/agenda-sort/agenda-sort.component.html]: Result: You can just use `item.getTitle()`.

File: client/src/app/shared/components/sorting-tree/sorting-tree.component.ts [hunk 4]
```diff
@@ -553,20 +671,20 @@ export class SortingTreeComponent<T extends Identifiable & Displayable> implemen
...
-        nextLevel: number,
-        verticalMove: Direction.UPWARDS | Direction.DOWNWARDS | Direction.NOWAY
-    ): void {
+    private moveItemToTree(node: FlatNode<T>, previousIndex: number, nextIndex: number, nextLevel: number): void {
+        let verticalMove =
+            previousIndex < nextIndex
+                ? Direction.DOWNWARDS
+                : previousIndex > nextIndex
+                ? Direction.UPWARDS
+                : Direction.NOWAY;
```
Comment (tsiegleauq) [inline client/src/app/shared/components/sorting-tree/sorting-tree.component.ts]: The usage of one ternary operator for a statement is often okay. If you require two, just take `if-else`.

**Comment by tsiegleauq** [comment]
Fixed travis issues and some a little coding style. Since other features depend on this, it should be mergable now

**Pull request merged**